### PR TITLE
Add class for checking well convergence

### PR DESCRIPTION
**Pull request by akva2**
More well code splitting.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
jenkins build this please

File: opm/simulators/wells/WellConvergence.cpp [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        case Well::InjectorCMode::RESV:
+            ctrltype = CR::WellFailure::Type::ControlRate;
+            control_tolerance = tolerances.rates;
+            break;
+        case Well::InjectorCMode::GRUP:
+            ctrltype = CR::WellFailure::Type::ControlRate;
+            control_tolerance = tolerances.grup;
+            break;
+        default:
+            OPM_DEFLOG_THROW(std::runtime_error, "Unknown well control control types for well " << well_.name(), deferred_logger);
```
Comment (bska) [inline opm/simulators/wells/WellConvergence.cpp:72]: We should include `<stdexcept>` to bring a declaration of `std::runtime_error` into scope.

File: opm/simulators/wells/WellConvergence.cpp [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        }
+    }
+
+    const int dummy_component = -1;
+    if (std::isnan(well_control_residual)) {
```
Comment (bska) [inline opm/simulators/wells/WellConvergence.cpp:106]: We must include `<cmath>` in order to use `std::isnan()`.

**Review by bska — COMMENTED** [review]
Just a couple of missing headers and then I'm ready to merge this.

File: opm/simulators/wells/WellConvergence.cpp (hunk 1 above)
Comment (akva2) [reply opm/simulators/wells/WellConvergence.cpp:75]: done.

File: opm/simulators/wells/WellConvergence.cpp (hunk 2 above)
Comment (akva2) [reply opm/simulators/wells/WellConvergence.cpp:109]: done.

**Comment by akva2** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks a lot for the updates.  I'll merge this into the master branch now.

**Pull request merged**